src/pocketmine/tile/Spawnable.php

```diff
@@ -96,7 +96,9 @@ protected function onChanged(){
 	 *
 	 * @param CompoundTag $nbt
 	 */
-	abstract public function addAdditionalSpawnData(CompoundTag $nbt);
+	public function addAdditionalSpawnData(CompoundTag $nbt){
+
```

Comment:
I would prefer this be explicitly overridden in EnderChest to make the fact clear, otherwise this may be unintentionally missed for other things.